### ISSUE TITLE
content-trust=false option fails to work correctly

### ISSUE THREAD

**Issue by tdejneka** [home-assistant/supervisor#2935]
### Describe the issue you are experiencing

- Prior to version 2021.5.4, setting `content-trust=false` allowed for Supervisor source code modifications. The instance was simply flagged as `unsupported`.

- In version 2021.5.4, the same setting results in the instance flagged as `unsupported` _and_ `unhealthy`. In other words, regardless of whether `content-trust` is set to `true` or `false`, it produces the same result when it detects source code modifications.

The `content-trust` option was suggested to me by frenck when I asked for a way to allow for modifying Supervisor's source code (a trivial patch that changes two integer values to reduce the polling frequency of connectivity/version checks).
https://github.com/home-assistant/supervisor/issues/2863#issuecomment-835604541

As far back as 2021.4.3, it worked perfectly but now, in 2021.5.4, it makes no difference if `content-trust` is `true` or `false`, the result is the same when it detects Supervisor source code modifications.

### What is the used version of the Supervisor?

supervisor-2021.05.4

### What type of installation are you running?

Home Assistant Supervised

### Which operating system are you running on?

Debian

### What is the version of your installed operating system?

Debian GNU/Linux 10 (buster)

### What version of Home Assistant Core is installed?

core-2021.6.0

### Steps to reproduce the issue

1. Make a trivial change to Supervisor's source code. For example, in `tasks.py` change the integer value on [line 50](https://github.com/home-assistant/supervisor/blob/64c59d0fe92bbc48f206cfaff3ed2709099c7a6a/supervisor/misc/tasks.py#L50) from `30` to `300`. On [line 456](https://github.com/home-assistant/supervisor/blob/64c59d0fe92bbc48f206cfaff3ed2709099c7a6a/supervisor/misc/tasks.py#L456), change it from `600` to `3600`.

2.  Restart Supervisor. The log will report the detected source code modifications and the instance will be flagged as `unsupported` and `unhealthy`.

3. Execute: `sudo ha supervisor options --content-trust=false`

4. Restart Supervisor. The log fails to mention anything about `content-trust` being disabled and the instance will still be flagged as `unsupported` and `unhealthy` (prior to 2021.5.4 the log would mention that content-trust was disabled and the instance would only be flagged as `unsupported`).

### Anything in the Supervisor logs that might be useful for us?

```txt
Log supplied in the next section (see below).
```

### Additional information

When the following command is executed:
```text
sudo ha supervisor options --content-trust=false
```
Supervisor's log reports:

```text
21-06-03 18:16:53 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state CoreState.RUNNING
21-06-03 18:16:53 WARNING (MainThread) [supervisor.resolution.evaluations.source_mods] Disabled content-trust, skipping evaluation
21-06-03 18:16:53 INFO (MainThread) [supervisor.resolution.evaluations.base] Clearing UnsupportedReason.SOURCE_MODS as reason for unsupported
21-06-03 18:16:53 WARNING (MainThread) [supervisor.resolution.evaluations.base] System run with disabled trusted content security. (more-info: https://www.home-assistant.io/more-info/unsupported/content_trust)
21-06-03 18:16:53 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
```
The log's third line suggests Supervisor acknowledges `content-trust=false` and it has cleared the `unsupported` flag. However, the UI shows the instance is still flagged as `unsupported` and `unhealthy`.
![Screenshot from 2021-06-03 14-18-50](https://user-images.githubusercontent.com/44732909/120697997-6d14ca80-c47c-11eb-931a-24879bf1eb23.png)

In addition, _restarting_ Supervisor doesn't clear either of the two flags. In fact, after a restart, Supervisor's log implies it has _forgotten_ about `content-trust=false` because there's no reference to it in the log (prior versions would mention content-trust was disabled on startup) and it flags the instance the moment it detects the source code changes.

Fifth line from the bottom of the log:
```test
21-06-03 18:27:41 WARNING (MainThread) [supervisor.resolution.evaluations.base] System detect unauthorized source code modifications. (more-info: https://www.home-assistant.io/more-info/unsupported/source_mods)
```

Full log after restart:
```text
21-06-03 18:27:04 INFO (MainThread) [supervisor.misc.scheduler] Shutting down scheduled tasks
21-06-03 18:27:04 INFO (MainThread) [supervisor.api] Stopping API on 172.30.32.2
21-06-03 18:27:04 INFO (MainThread) [supervisor.hardware.monitor] Stopped Supervisor hardware monitor
21-06-03 18:27:04 INFO (MainThread) [supervisor.core] Supervisor is down - 100
21-06-03 18:27:04 INFO (MainThread) [__main__] Closing Supervisor
21-06-03 18:27:11 INFO (MainThread) [__main__] Initializing Supervisor setup
21-06-03 18:27:12 INFO (MainThread) [supervisor.bootstrap] Initializing Supervisor Sentry
21-06-03 18:27:12 INFO (MainThread) [supervisor.bootstrap] Seting up coresys for machine: raspberrypi3
21-06-03 18:27:12 INFO (SyncWorker_0) [supervisor.docker.supervisor] Attaching to Supervisor homeassistant/aarch64-hassio-supervisor with version 2021.05.4
21-06-03 18:27:12 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
21-06-03 18:27:13 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state CoreState.INITIALIZE
21-06-03 18:27:13 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
21-06-03 18:27:13 INFO (MainThread) [__main__] Setting up Supervisor
21-06-03 18:27:13 INFO (MainThread) [supervisor.api] Starting API on 172.30.32.2
21-06-03 18:27:13 INFO (MainThread) [supervisor.hardware.monitor] Started Supervisor hardware monitor
21-06-03 18:27:21 WARNING (MainThread) [supervisor.dbus.rauc] Host has no rauc support. OTA updates have been disabled.
21-06-03 18:27:21 INFO (MainThread) [supervisor.host.info] Updating local host information
21-06-03 18:27:21 INFO (MainThread) [supervisor.host.services] Updating service information
21-06-03 18:27:21 INFO (MainThread) [supervisor.host.network] Updating local network information
21-06-03 18:27:26 INFO (MainThread) [supervisor.host.sound] Updating PulseAudio information
21-06-03 18:27:26 INFO (MainThread) [supervisor.host] Host information reload completed
21-06-03 18:27:26 INFO (MainThread) [supervisor.host.apparmor] Loading AppArmor Profiles: {'hassio-supervisor'}
21-06-03 18:27:26 INFO (MainThread) [supervisor.host.services] Reloading local service hassio-apparmor.service
21-06-03 18:27:27 INFO (SyncWorker_0) [supervisor.docker.interface] Attaching to homeassistant/aarch64-hassio-dns with version 2021.04.0
21-06-03 18:27:27 INFO (MainThread) [supervisor.plugins.dns] Updated /etc/resolv.conf
21-06-03 18:27:27 INFO (SyncWorker_0) [supervisor.docker.interface] Attaching to homeassistant/aarch64-hassio-audio with version 2021.04.0
21-06-03 18:27:27 INFO (SyncWorker_0) [supervisor.docker.interface] Attaching to homeassistant/aarch64-hassio-cli with version 2021.05.1
21-06-03 18:27:27 INFO (SyncWorker_0) [supervisor.docker.interface] Attaching to homeassistant/aarch64-hassio-observer with version 2020.10.1
21-06-03 18:27:27 INFO (SyncWorker_0) [supervisor.docker.interface] Attaching to homeassistant/aarch64-hassio-multicast with version 2021.04.0
21-06-03 18:27:27 INFO (MainThread) [supervisor.updater] Fetching update data from https://version.home-assistant.io/stable.json
21-06-03 18:27:29 INFO (MainThread) [supervisor.homeassistant.secrets] Loaded 2 Home Assistant secrets
21-06-03 18:27:29 INFO (SyncWorker_3) [supervisor.docker.interface] Attaching to homeassistant/raspberrypi3-homeassistant with version 2021.6.0
21-06-03 18:27:29 INFO (MainThread) [supervisor.hassos] No Home Assistant Operating System found
21-06-03 18:27:32 INFO (MainThread) [supervisor.store.git] Loading add-on /data/addons/core repository
21-06-03 18:27:32 INFO (MainThread) [supervisor.store.git] Loading add-on /data/addons/git/a0d7b954 repository
21-06-03 18:27:37 INFO (MainThread) [supervisor.store] Loading add-ons from store: 63 all - 63 new - 0 remove
21-06-03 18:27:37 INFO (MainThread) [supervisor.addons] Found 3 installed add-ons
21-06-03 18:27:37 INFO (SyncWorker_2) [supervisor.docker.interface] Attaching to homeassistant/armv7-addon-samba with version 9.5.0
21-06-03 18:27:37 INFO (SyncWorker_1) [supervisor.docker.interface] Attaching to ghcr.io/hassio-addons/portainer/armv7 with version 1.5.0
21-06-03 18:27:37 INFO (SyncWorker_0) [supervisor.docker.interface] Attaching to homeassistant/armv7-addon-configurator with version 5.3.1
21-06-03 18:27:37 INFO (MainThread) [supervisor.snapshots] Found 10 snapshot files
21-06-03 18:27:37 INFO (MainThread) [supervisor.discovery] Loaded 0 messages
21-06-03 18:27:37 INFO (MainThread) [supervisor.ingress] Loaded 1 ingress sessions
21-06-03 18:27:37 INFO (MainThread) [supervisor.resolution.check] Starting system checks with state CoreState.SETUP
21-06-03 18:27:37 WARNING (MainThread) [supervisor.resolution.check] Skipping disabled check addon_pwned
21-06-03 18:27:37 INFO (MainThread) [supervisor.resolution.check] System checks complete
21-06-03 18:27:37 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state CoreState.SETUP
21-06-03 18:27:37 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
21-06-03 18:27:37 INFO (MainThread) [supervisor.jobs] 'ResolutionFixup.run_autofix' blocked from execution, system is not running - CoreState.SETUP
21-06-03 18:27:37 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state CoreState.SETUP
21-06-03 18:27:38 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
21-06-03 18:27:38 INFO (MainThread) [__main__] Running Supervisor
21-06-03 18:27:38 INFO (MainThread) [supervisor.jobs] 'HassOS.mark_healthy' blocked from execution, no Home Assistant OS available
21-06-03 18:27:38 INFO (MainThread) [supervisor.addons] Phase 'AddonStartup.INITIALIZE' starting 0 add-ons
21-06-03 18:27:38 INFO (MainThread) [supervisor.core] Supervisor reboot detected
21-06-03 18:27:38 INFO (MainThread) [supervisor.misc.tasks] All core tasks are scheduled
21-06-03 18:27:38 INFO (MainThread) [supervisor.core] Supervisor is up and running
21-06-03 18:27:38 INFO (MainThread) [supervisor.host.info] Updating local host information
21-06-03 18:27:38 INFO (MainThread) [supervisor.resolution.check] Starting system checks with state CoreState.RUNNING
21-06-03 18:27:38 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.FREE_SPACE/ContextType.SYSTEM
21-06-03 18:27:38 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.SECURITY/ContextType.CORE
21-06-03 18:27:38 WARNING (MainThread) [supervisor.resolution.check] Skipping disabled check addon_pwned
21-06-03 18:27:38 INFO (MainThread) [supervisor.resolution.check] System checks complete
21-06-03 18:27:38 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state CoreState.RUNNING
21-06-03 18:27:38 INFO (MainThread) [supervisor.host.services] Updating service information
21-06-03 18:27:38 INFO (MainThread) [supervisor.host.network] Updating local network information
21-06-03 18:27:41 WARNING (MainThread) [supervisor.resolution.evaluations.base] System detect unauthorized source code modifications. (more-info: https://www.home-assistant.io/more-info/unsupported/source_mods)
21-06-03 18:27:41 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
21-06-03 18:27:41 INFO (MainThread) [supervisor.jobs] 'ResolutionFixup.run_autofix' blocked from execution, system is not healthy
21-06-03 18:27:44 INFO (MainThread) [supervisor.host.sound] Updating PulseAudio information
21-06-03 18:27:44 INFO (MainThread) [supervisor.host] Host information reload completed
```

**Comment by ludeeus** [comment]
You are confusing content trust and source_mods.
Checks can be disabled with the resolution API <https://developers.home-assistant.io/docs/api/supervisor/endpoints#resolution>

**Comment by tdejneka** [comment]
Thanks for taking the time to reply to my Issue.

> You are confusing content trust and source_mods.

OK, I'll accept that then what is the correct way to disable checking for source_mods? You directed me to the resolution API but `/resolution/check/<slug>/options` provides no explanation for what is accepted in `<slug>`.

For that matter, neither does the CLI. Here's what it says for `ha resolution check options --help`

>This command allows to apply options to an specific check managed by the system.
>
>Usage:
>  ha resolution check options [flags]
>
>Aliases:
>  options, option, opt, opts, op
>
>Examples:
>
>  ha resolution check options [slug]

There is nothing to explain what is acceptable for `slug`.

Given that I couldn't figure out how to use the command correctly, I modified the contents of `resolution.json` and changed the entry for `core_security` to `false` then rebooted the host machine.

```json
{
  "checks": {
    "addon_pwned": {
      "enabled": false
    },
    "core_security": {
      "enabled": false
    },
    "free_space": {
      "enabled": true
    }
  }
}
```

That was accepted and reported in Supervisor's log:
```text
21-07-08 12:56:08 WARNING (MainThread) [supervisor.resolution.check] Skipping disabled check core_security
```

However, I failed yet again because that option doesn't appear to control the detection of source-code modifications. Supervisor promptly flagged the instance as 'unhealthy' when it detected the source-code modification I had made:
```text
21-07-08 12:56:12 WARNING (MainThread) [supervisor.resolution.evaluations.base] System detect unauthorized source code modifications. (more-info: https://www.home-assistant.io/more-info/unsupported/source_mods)
```
As a consequence of being "unhealthy" it blocked updating Add-ons, etc.

Can you help me set the correct option to suppress the detection of mods made to Supervisor's source code? I can't figure out how to do it with the CLI.

**Comment by tdejneka** [comment]
I am led to conclude that you have closed this Issue prematurely. You closed it with the following statement:

> Checks can be disabled with the resolution API

Based on my experiments using the CLI, there are only three resolution options that can be disabled and **none of them** disable source-code checking.

1. core_security
2. free_space
3. addon_pwned

This Issue is specifically for the ability to disable Supervisor's source-code checking. It was available in a previous version of Supervisor but the method no longer works. Clearly, something changed. It's either:

1. A bug.
2. Replaced by a new method to disable source-code checking.
3. The ability to disable source-code checking has been eliminated.


If it's not a bug, can you describe the new way to do it, or confirm that it has been eliminated?

**Comment by tdejneka** [comment]
The favor of a reply is requested from anyone on the Supervisor development team.

Line 42 of `source_mods.py` indicates that it will skip the check of source-code modifications if content-trust is **disabled**. However, it fails to work as advertised. Disabling content-trust via the CLI still results in a source-code check. The fault lies _somewhere_, if not in Supervisor's code then in the CLI for failing to truly disable content-trust.

https://github.com/home-assistant/supervisor/blob/85e7f817e6b445a3e06bec1fb7e50eccccc77db2/supervisor/resolution/evaluations/source_mods.py#L42

**Comment by ludeeus** [comment]
I'm sorry, this was moved...
Use `curl -sSL -H "Authorization: Bearer $SUPERVISOR_TOKEN" -X POST http://supervisor/security/options -d '{"content_tru
st": false}'` to disable from any of the SSH add-ons.

**Comment by tdejneka** [comment]
Thank you for taking the time to reply. I appreciate it.

Using the Portainer add_on to gain access to the console of the `addon_core_samba` container, I executed your suggested command:

```text
curl -sSL -H "Authorization: Bearer $SUPERVISOR_TOKEN" -X POST http://supervisor/security/options -d '{"content_tru st": false}'
```

However, the response was:
```text
403: Forbidden
```
I received the same response when I repeated the test in the `addon_core_configurator` container. When executed from within `hassio_supervisor` container the response was:
```text
401: Unauthorized
```
I used `echo %SUPERVISOR_TOKEN` to confirm it produces a long identifier.

I double-checked the API documentation and the command shown in the [Security section](https://developers.home-assistant.io/docs/api/supervisor/endpoints/#security) indicates the URL is:
```text
http://security/options
```

I modified the command as follows and executed it within the `addon_core_samba` container:
```text
curl -sSL -H "Authorization: Bearer $SUPERVISOR_TOKEN" -X POST http://security/options -d '{"content_tru st": false}'
```

However, after a brief delay of about 5 seconds, the response was:
```text
curl: (6) Could not resolve host: security
```

I repeated the tests on a second instance of Home Assistant Supervised with the same results (both use 2021.6.6 with Supervisor 2021.06.8).

I don't know what else to change and need your help to make the command work properly.

**Comment by ludeeus** [comment]
1) samba is not one of the SSH add-ons
2) `http://security/` is not `http://supervisor/security`

**Comment by tdejneka** [comment]
My mistake! I misunderstood the instructions in your post.

I don't have the SSH Add-on installed so I'll take care of that first then retry your suggested command.

**Comment by tdejneka** [comment]
I installed the SSH & Web Terminal Add-on, executed the command and it reported the following error (apologies for the screenshot; it doesn't let me copy onscreen text).

![Screenshot from 2021-07-12 11-32-43](https://user-images.githubusercontent.com/44732909/125315453-25981e80-e305-11eb-8905-f5809ebdf15e.png)

However, the error message is informative and made me realize the suggested command contains a small typo. There's an added  space in "content_tru st"!  I removed it, executed the command and this time it worked and responded with:
```text
{"result": "ok", "data":{}}
```
The change was registered immediately in Supervisor's log:
```text
21-07-12 11:36:37 WARNING (MainThread) [supervisor.resolution.evaluations.source_mods] Disabled content-trust, skipping evaluation
21-07-12 11:36:37 WARNING (MainThread) [supervisor.resolution.evaluations.base] System run with disabled trusted content security. (more-info: https://www.home-assistant.io/more-info/unsupported/content_trust)
```
The system's status was changed to Unsupported (which was expected) but now remains Healthy.

I guess the only remaining question is the original one; why did this fail to work:
```text
ha supervisor options --content-trust=false
```
If there's no available answer, you have provided me with an alternative that does work.
